---
title: "00_wgs-clean-data-baseline-varibales"
author: "Shawn Zamani"
date: "`r Sys.Date()`"
output: html_document
editor_options: 
  chunk_output_type: console
---

# Objective

Generate descriptive statistics tables (Table 1) for the following cohorts:
- OAC WGS cases, 710
# Setup

```{r setup}
library(tidyverse) 
library(table1)
library(kableExtra)
```

```{r read_data}
load('data/original/LabKey-occams_pt-tibble_20220617.RData')
overall_cohort <- occams_pt_20220617

filtered_cohort <- overall_cohort %>%
  # group_by(ID) %>% 
  # filter(n()==1) %>% 
  # ungroup(.) %>% 
  filter(!is.na(DI.ageAtDiagnosis)) %>% 
  filter(!is.na(DI.PatientGender)) %>% 
  distinct(ID, .keep_all = TRUE)

OAC_710_cohort <-
  vroom::vroom('data/original/wgs/case_spec_dna_occams_oac_normal.710.csv', col_names = FALSE) %>%
  transmute(
    ID = str_replace(X1, 'OC/', 'OCCAMS/'),
    spec_oac = X2,
    dna_oac = X3,
    dna_normal = X4
  ) %>% 
  select(-spec_oac) %>% 
  rename(abnormal_DNA_ID = dna_oac,
         normal_DNA_ID = dna_normal) %>% 
  unite(col = DNA_ID, abnormal_DNA_ID, normal_DNA_ID, remove = FALSE, sep = '_vs_') %>% 
  mutate(cohort = '710')


oac_gel_cohort <-
  vroom::vroom('data/original/wgs/gel_submission_metadata_ginny_BO_OAC.csv.csv', col_names = TRUE) %>% 
  rename(abnormal_DNA_ID = abnormal_ID) %>% 
  mutate(cohort = 'GEL') %>% 
  filter(grepl('OCCAMS', ID)) %>% 
  distinct(ID, .keep_all = T)


oac_gel_710_combined_cohort <- bind_rows(OAC_710_cohort, oac_gel_cohort) %>% relocate(.after = ID, cohort)

occams_molecular_thesis_cohort <- left_join(oac_gel_710_combined_cohort, filtered_cohort, "ID")%>%  relocate(c(DNA_ID, abnormal_DNA_ID, normal_DNA_ID), .after = ID)

# occams_molecular_thesis_cohort <-
  # inner_join(filtered_cohort, oac_gel_710_combined_cohort, "ID")

# occams_molecular_thesis_cohort <- inner_join(filtered_cohort, OAC_710_cohort, "ID")
# occams_molecular_thesis_cohort <- inner_join(filtered_cohort, OAC_GEL_cohort %>% distinct(ID), "ID")

# table(overall_cohort$EX.BarrettsOesophagusUndergoingSurveillance, useNA = 'always')

```

# Clean data

## pathology

```{r pathology, include=FALSE}
occams_molecular_thesis_cohort_clean_path <- occams_molecular_thesis_cohort %>%
  select(
    ID,
    DNA_ID, 
    abnormal_DNA_ID, 
    normal_DNA_ID,
    StudySite,
    cohort,
    Weeks.Survival.c,
    RD.DiagnosisDate.c,
    DI.ageAtDiagnosis,
    DI.PatientGender,
    DI.Ethnic.Category,
    EX.PatientOnAcidSuppressant,
    EX.SymptomaticOnAcidSuppressant,
    EX.RefluxFrequencyOfSymptoms,
    EX.DurationOfRefluxSymptoms, #add never
    EX.IsSmoker,
    EX.AverageCigarrettesPerDay, #add logic to create numeric measures
    EX.SmokingDurationYears,
    EX.UnitsPerWeekInTotal,
    EX.HeavyDrinker, 
    EX.CurrentWeightKG,
    EX.CurrentHeightCM,
    EX.5YearsAgoWeightKG,
    EX.BarrettsOesophagusUndergoingSurveillance,
    EX.BarrettsOesophagusDateDiagnosed,
    EX.FamilyHistoryOfOesophagoGastricCancer,
    contains('unitsperweek'),
    contains('EX.NSAIDs'),
    contains('EX.ProtonPump'),
    contains('EX.HistamineType2'), #high missingness, not useful to clean
    contains('EX.OverTheCounter'),
    contains('EX.Asprin'),
    RP.Histology,
    # PS.Histology,
    # PS.TStage.PrimaryTumour.FinalPretreatmentStaging,
    # PS.NStage.PrimaryTumour.FinalPretreatmentStaging.TNM7,
    # PS.MStage.PrimaryTumour.FinalPretreatmentStaging,
    RP.TStage.PrimaryTumour,
    RP.Nstage.RP.TNM7,
    RP.MStage.DistantMetastasis,
    RP.MandardScoreForResponse,
    RP.TumourGradingDifferentiationStatus,
    RP.TumourDifferentiation.c,
    RP.Tumour.Size.length.mm,
    RP.SiewertClassification,
    RP.Location,
    RP.PostOpPathCombinedTumourSite,
    # below is for alvin ecdna paper nov 2023
    TR.NeoAdj.ChemotherapyRegimenEpirubicin,
    TR.NeoAdj.ChemotherapyRegimenCisplatin,
    TR.NeoAdj.ChemotherapyRegimenOxaliplatin,
    TR.NeoAdj.ChemotherapyRegimen5FU,
    TR.NeoAdj.ChemotherapyRegimenCapecitabine,

    # RP.CircumferentialResectionMargin,
    # RP.LymphoVascularInvasion,
    # TR.CurativeTreatmentModality,
    # ST.Procedure,
    ST.SurgeryPerformed.c,
    RP.BarrettsAdjacentToTumour.c,
    # RP.TumourDifferentiation.c,
    Patient.Died.c,
    ends_with(".c")
  ) %>%
  mutate(
    Phenotype = case_when(
      RP.BarrettsAdjacentToTumour.c == "yes" ~ "BE/IM EAC",
      RP.BarrettsAdjacentToTumour.c == "no" ~ "Non-BE/IM EAC",
      is.na(RP.BarrettsAdjacentToTumour.c) ~ "Unknown BE/IM EAC"
    )
  ) %>% 
  mutate(
  pTStage_Prm_Tmr_f = factor(
    RP.TStage.PrimaryTumour,
    levels = c(
      'Tx',
      'T0',
      'Tis',
      'T1',
      'T1a',
      'T1b',
      'T2',
      'T3',
      'T4',
      'T4a',
      'T4b'
    ),
    ordered = TRUE
  ),
  
  RP.TumourDifferentiation.c = factor(
    RP.TumourDifferentiation.c,
    levels = c('poor',
               'moderate',
               'well'),
    ordered = TRUE
  ),
  
  pTStage_four_f = factor(
    case_when(
      RP.TStage.PrimaryTumour %in% c("T1", "T1a", "T1b") ~ "T1",
      RP.TStage.PrimaryTumour == "T2" ~ "T2",
      RP.TStage.PrimaryTumour == "T3" ~ "T3",
      RP.TStage.PrimaryTumour %in% c("T4", "T4a", "T4b") ~ "T4"
    ),
    levels = c("T1",
               "T2",
               "T3",
               "T4"),
    ordered = TRUE
  ),
  
  pNStage_Prm_Tmr_f = factor(
    RP.Nstage.RP.TNM7,
    levels = c('Nx',
               'N0',
               'N1',
               'N2',
               'N3'),
    ordered = TRUE
  ),
  
  pMStage_Dist_Metas_f = factor(
    RP.MStage.DistantMetastasis,
    levels = c('Mx',
               'M0',
               'M1'),
    ordered = TRUE
  )
) %>%
  
  mutate(
    pTNM_c = case_when((
      (
        pTStage_Prm_Tmr_f %in% c('T1', 'T1a', 'T1b') &
          pNStage_Prm_Tmr_f == 'N0'
      ) |
        (
          pTStage_Prm_Tmr_f == 'T2' &
            pNStage_Prm_Tmr_f == 'N0' &
            RP.TumourDifferentiation.c != 'poor'
        )
    ) ~ "I",
    (
      (
        pTStage_Prm_Tmr_f == 'T2' &
          pNStage_Prm_Tmr_f == 'N0' &
          RP.TumourDifferentiation.c == 'poor'
      ) |
        (pTStage_Prm_Tmr_f == 'T3' & pNStage_Prm_Tmr_f == 'N0') |
        (
          pTStage_Prm_Tmr_f %in% c('T1', 'T1a', 'T1b', 'T2') &
            pNStage_Prm_Tmr_f == 'N1'
        )
    ) ~ "II",
    (
      (
        pTStage_Prm_Tmr_f %in% c('T1', 'T1a', 'T1b', 'T2') &
          pNStage_Prm_Tmr_f == 'N2'
      ) |
        (pTStage_Prm_Tmr_f == 'T3' & pNStage_Prm_Tmr_f == 'N1') |
        (
          pTStage_Prm_Tmr_f %in% c('T4', 'T4a', 'T4b') &
            pNStage_Prm_Tmr_f == 'N0'
        ) |
        (pTStage_Prm_Tmr_f == 'T3' & pNStage_Prm_Tmr_f == 'N2') |
        (
          pTStage_Prm_Tmr_f %in% c('T4', 'T4a', 'T4b') &
            pNStage_Prm_Tmr_f %in% c(NA, 'N1', 'N2')
        ) |
        (pNStage_Prm_Tmr_f == 'N3')
    ) ~ "III"),
    
    pTNM_f = ifelse(pMStage_Dist_Metas_f %in% c(NA, 'Mx', 'M0'), pTNM_c, "IV"),
    #if metastisized, then TNM IV
    
    pTNM_f = factor(
      pTNM_f,
      levels = c("I", "II", "III", "IV"),
      ordered = TRUE
    ),
    
    pTNM_Split_f = ifelse(
      pTNM_f %in% c('I', "II"),
      "[pTNM I/II]",
      ifelse(pTNM_f %in% c('III', "IV"), "[pTNM III/IV]", NA)
    ),
    
    pTNM_Split_f = factor(
      pTNM_Split_f,
      levels = c("[pTNM I/II]", "[pTNM III/IV]"),
      ordered = TRUE
    ),
    
    pTStage_Split_f = factor(
      case_when(
        pTStage_four_f %in% c("T1", "T2") ~ "T1/T2",
        pTStage_four_f %in% c("T3", "T4") ~ "T3/T4"
      ),
      levels = c("T1/T2",
                 "T3/T4"),
      ordered = TRUE
    ),
    Tumor_Length_CM_n = as.numeric(RP.Tumour.Size.length.mm / 10)
  ) %>% 
    mutate(RP.BarrettsAdjacentToTumour.c = case_when(
                     is.na(RP.BarrettsAdjacentToTumour.c) ~ 'Unknown',
                   TRUE ~ as.character(RP.BarrettsAdjacentToTumour.c))) %>% 
  mutate(Years_Surv = Weeks.Survival.c / 52) %>%
  mutate(Death = case_when(Patient.Died.c == 'yes' ~ 1, TRUE ~ 0))
```

## epidemiology

Common variables:

-   Age
-   Gender
-   BMI
-   Ethnicity
-   Smoking
-   NSAID
-   ACID_SUPPRESSANT
-   Duration_Reflux_Symptoms
-   Phenotype

```{r risk_factors, include=FALSE}
occams_molecular_thesis_cohort_clean_path_epid <- occams_molecular_thesis_cohort_clean_path %>%
  mutate(age_group = factor(
    case_when(
      DI.ageAtDiagnosis < 50 ~ "< 50 years old",
      DI.ageAtDiagnosis >= 50 &
        DI.ageAtDiagnosis < 60 ~ "50 - 59 years old",
      DI.ageAtDiagnosis >= 60 &
        DI.ageAtDiagnosis < 70 ~ "60 - 69 years old",
      DI.ageAtDiagnosis >= 70 ~ "70+ years old"
      # TRUE ~ age_at_be_diag_or_baseline
      # age_at_be_diag_or_baseline >= 80 ~ "80+ years old"
    ),
    # levels = c(
    #   "< 50 years old",
    #   "50 - 59 years old",
    #   "60 - 69 years old",
    #   "70+ years old"
    #   # "80+ years old"
    # ),
    ordered = TRUE
  )) %>%
  mutate(
    gender = case_when(
      DI.PatientGender == "female" ~ "F",
      DI.PatientGender == "male" ~ "M",
      TRUE ~ DI.PatientGender
    )
  ) %>% 
  mutate(EX.IsSmoker = case_when(
    (
      is.na(EX.IsSmoker) &
        EX.AverageCigarrettesPerDay %in% c('-1', '0')
    ) ~ 'never',
    TRUE ~ EX.IsSmoker
  )) %>%
  mutate(
    smoking_CPD = case_when(
      # clean the CPD
      EX.AverageCigarrettesPerDay %in% c('-1') ~ NA_character_,
      EX.AverageCigarrettesPerDay %in% c('-0') ~ '0',
      EX.AverageCigarrettesPerDay %in% c('0.5') ~ '1',
      EX.AverageCigarrettesPerDay %in% c('1-2', '2 CIGARS', '2 - cigars/day') ~ '2',
      EX.AverageCigarrettesPerDay %in% c('2-3', '2/3', '23') ~ '3',
      EX.AverageCigarrettesPerDay %in% c('3-4') ~ '3',
      EX.AverageCigarrettesPerDay %in% c('40 packs a year, 2 per day') ~ '2',
      EX.AverageCigarrettesPerDay %in% c('4-5', '5 cigar rolls', 'Occasional') ~ '5',
      EX.AverageCigarrettesPerDay %in% c('5-6') ~ '6',
      EX.AverageCigarrettesPerDay %in% c('5-10') ~ '7',
      EX.AverageCigarrettesPerDay %in% c('5-7') ~ '7',
      EX.AverageCigarrettesPerDay %in% c('7-10') ~ '8',
      EX.AverageCigarrettesPerDay %in% c('5-12', '9') ~ '10',
      EX.AverageCigarrettesPerDay %in% c('10-12') ~ '11',
      EX.AverageCigarrettesPerDay %in% c('5-20') ~ '12',
      EX.AverageCigarrettesPerDay %in% c('10-15', '10/15') ~ '13',
      EX.AverageCigarrettesPerDay %in% c('12-15') ~ '13',
      EX.AverageCigarrettesPerDay %in% c('5-25') ~ '15',
      EX.AverageCigarrettesPerDay %in% c('10-20') ~ '15',
      EX.AverageCigarrettesPerDay %in% c('14/15') ~ '15',
      EX.AverageCigarrettesPerDay %in% c('15-20') ~ '18',
      EX.AverageCigarrettesPerDay %in% c('20, stopped 30yrs ago') ~ '20',
      EX.AverageCigarrettesPerDay %in% c('20-30') ~ '25',
      EX.AverageCigarrettesPerDay %in% c('25-30') ~ '27',
      EX.AverageCigarrettesPerDay %in% c('20-40') ~ '30',
      EX.AverageCigarrettesPerDay %in% c('30-40') ~ '35',
      EX.AverageCigarrettesPerDay %in% c('60-80') ~ '70',
      TRUE ~ EX.AverageCigarrettesPerDay
    )
  ) %>%
  mutate(smoking_CPD = as.double(smoking_CPD)) %>%
  mutate(smoking_CPD = case_when(smoking_CPD <= 0 ~ NA_integer_,
                                 TRUE ~ as.integer(smoking_CPD))) %>%
  mutate(EX.IsSmoker = case_when(
    (EX.IsSmoker %in% c('never', NA) & smoking_CPD > 0 ) ~ 'former',
    TRUE ~ EX.IsSmoker)) %>% 
  mutate(smoking_NPY = (smoking_CPD / 20) * (EX.SmokingDurationYears)) %>% # calculate pack-years
  mutate(smoking_CPD = na_if(smoking_CPD, 0)) %>%
  mutate(smoking_duration_yrs = na_if(EX.SmokingDurationYears, 0.000)) %>%
  mutate(smoking_NPY = na_if(smoking_NPY, 0)) %>%
  mutate(
  EX.RefluxFrequencyOfSymptoms_f = factor(
    EX.RefluxFrequencyOfSymptoms,
    levels = c("never", "sometimes", "often","daily"),
    ordered = TRUE
  ),
  
  EX.IsSmoker_f = factor(
    EX.IsSmoker, levels = c('never', 'current', 'former'),
    ordered = TRUE
  ),
  
  Smoker_Ever = factor(case_when(
    EX.IsSmoker %in% c('current', 'former') ~ 'ever',
    EX.IsSmoker_f == 'never' ~ 'never'),
    levels = c('never', 'ever'),
    ordered = TRUE
  ),
  
  EX.DurationOfRefluxSymptoms_f = factor(
    EX.DurationOfRefluxSymptoms,
    levels = c("< 5 years", "5-10 years", "> 10 years"),
    ordered = TRUE
  ),

  EX.DurationOfRefluxSymptoms_f = factor(case_when(
    EX.RefluxFrequencyOfSymptoms_f == 'never' ~ 'never',
    EX.RefluxFrequencyOfSymptoms_f %in% c("sometimes", "often","daily") & is.na(EX.DurationOfRefluxSymptoms_f) ~ 'unknown',
    TRUE ~ EX.DurationOfRefluxSymptoms), 
    levels = c('never', '< 5 years', '5-10 years', '> 10 years', 'unknown'), 
    ordered = TRUE),
  
  EX.RefluxFrequencyOfSymptoms_f = factor(case_when(
    EX.DurationOfRefluxSymptoms_f %in%  c("< 5 years", "5-10 years", "> 10 years") & is.na(EX.RefluxFrequencyOfSymptoms_f) ~ 'unknown',
    TRUE ~ EX.RefluxFrequencyOfSymptoms), 
    levels = c("never", "sometimes", "often","daily", 'unknown'), 
    ordered = TRUE),

  Reflux_Ever_f = factor(case_when(
    EX.RefluxFrequencyOfSymptoms_f %in% c("daily", "often", "sometimes") |
      EX.DurationOfRefluxSymptoms_f %in% c("< 5 years", "5-10 years", "> 10 years") ~ "ever",
    EX.RefluxFrequencyOfSymptoms_f == "never" & is.na(EX.DurationOfRefluxSymptoms_f) ~ "never"),
    levels = c('never', 'ever'),
    ordered = TRUE
  ),
  
  EX.5YearsAgoWeightKG = ifelse(EX.5YearsAgoWeightKG <= 20.0, NA, EX.5YearsAgoWeightKG),
  
  EX.CurrentWeightKG = ifelse(EX.CurrentWeightKG <= 20.0, NA, EX.CurrentWeightKG),
  
  BMI_5yrs_Ago_n = as.double((EX.5YearsAgoWeightKG) / ((EX.CurrentHeightCM / 100.0) ^ 2)),
  
  BMI_n = as.double((EX.CurrentWeightKG) / ((EX.CurrentHeightCM/100.0) ^ 2)),
  
BMI_n = case_when(
  BMI_n == 0.0 ~ NA_integer_,
  BMI_n > 70.0 ~ NA_integer_,
  TRUE ~ as.integer(BMI_n)
),
  BMI_Diff_n = as.numeric(BMI_n - BMI_5yrs_Ago_n),
  
  BMI_Class_5yrs_Ago_f = factor(
    case_when(
      BMI_5yrs_Ago_n < 18.5 ~ "underweight",
      BMI_5yrs_Ago_n >= 18.5 & BMI_5yrs_Ago_n <= 24.9 ~ "normal",
      BMI_5yrs_Ago_n >= 25.0 & BMI_5yrs_Ago_n <= 29.9 ~ "overweight",
      BMI_5yrs_Ago_n >= 30.0 ~ "obese"
    ),
    levels = c("underweight", "normal", "overweight", "obese"),
    ordered = TRUE
  ), 
  
  BMI_Class_f = factor(
    case_when(
      BMI_n < 18.5 ~ "underweight",
      BMI_n >= 18.5 & BMI_n <= 24.9 ~ "normal",
      BMI_n >= 25.0 & BMI_n <= 29.9 ~ "overweight",
      BMI_n >= 30.0 ~ "obese"
    ),
    levels = c("underweight", "normal", "overweight", "obese"),
    ordered = TRUE
  ),
  
  BMI_Obese_f = factor(
    case_when(
      BMI_Class_f == 'obese' ~ 'obese',
      BMI_Class_f %in% c("underweight", "normal", "overweight") ~ 'not obese'
    ),      
    levels = c("not obese", "obese"),
    ordered = TRUE
  )
) %>%
  unite(
    'BMI_Class_5yrs_To_Dx_f',
    sep = " -> ",
    c(BMI_Class_5yrs_Ago_f, BMI_Class_f),
    remove = FALSE
  ) %>%
  mutate(Ethnicity_temp = na_if(DI.Ethnic.Category, 'Z')) %>% # Set ethnicity to NA if unknown
  mutate(Ethnicity_temp = na_if(Ethnicity_temp, '99')) %>%
  mutate(Ethnicity = case_when(
    Ethnicity_temp %in% c('A', 'B', 'C') ~ 'White',
    Ethnicity_temp %in% c('D', 'H', 'J', 'K', 'L', 'M', 'N', 'P', 'R', 'S') ~ 'Other'
  )) %>% 

  rowwise() %>% 
  mutate(EX.UnitsPerWeekInTotal = sum(EX.UnitsPerWeekOfWine, EX.UnitsPerWeekOfBeer, EX.UnitsPerWeekOfSpirits, na.rm = F))  %>%  
  mutate(EX.UnitsPerWeekInTotal = na_if(EX.UnitsPerWeekInTotal, 0))
```

## medications

-   Clean medication use variables
    -   PPI use and duration (years)
    -   NSAID use and duration (years)
    -   OTC acid suppressant use and duration (years)
-   clean up the durations by converting all measures into a single duartion with years as unit
  -   I have checked to see if I can use duration to convert NA frequncies into EVER - **this is not worth doing because there are 1 or 2 cases that have a non-zero duration but missing a frequency response**.
```{r med_clean, include=FALSE}

occams_molecular_thesis_cohort_clean_path_epid_meds <- occams_molecular_thesis_cohort_clean_path_epid %>%
  mutate(
    EX.NSAIDsDurationMonths_Yr = EX.NSAIDsDurationMonths / 12,
    EX.NSAIDsDurationWeeks_Yr = EX.NSAIDsDurationWeeks / 52,
    EX.NSAIDsDurationDays_Yr = EX.NSAIDsDurationDays / 365,
    
    EX.AsprinDurationMonths_Yr = EX.AsprinDurationMonths / 12,
    EX.AsprinDurationWeeks_Yr = EX.AsprinDurationWeeks / 52,
    EX.AsprinDurationDays_Yr = EX.AsprinDurationDays / 365, 
    
    EX.ProtonPumpInhibitorDurationMonths_Yr = EX.ProtonPumpInhibitorDurationMonths / 12,
    EX.ProtonPumpInhibitorDurationWeeks_Yr = EX.ProtonPumpInhibitorDurationWeeks / 52,
    EX.ProtonPumpInhibitorDurationDays_Yr = EX.ProtonPumpInhibitorDurationDays / 365,
    
    EX.OverTheCounterAcidSuppressantsDurationMonths_Yr = EX.OverTheCounterAcidSuppressantsDurationMonths / 12,
    EX.OverTheCounterAcidSuppressantsDurationWeeks_Yr = EX.OverTheCounterAcidSuppressantsDurationWeeks / 52,
    EX.OverTheCounterAcidSuppressantsDurationDays_Yr = EX.OverTheCounterAcidSuppressantsDurationDays / 365,
    
    EX.HistamineType2ReceptorAntagonistDurationMonths_Yr = EX.HistamineType2ReceptorAntagonistDurationMonths / 12,
    EX.HistamineType2ReceptorAntagonistDurationWeeks_Yr = EX.HistamineType2ReceptorAntagonistDurationWeeks / 52,
    EX.HistamineType2ReceptorAntagonistDurationDays_Yr = EX.HistamineType2ReceptorAntagonistDurationDays / 365,
    
  ) %>%
  rowwise() %>%
  mutate(
    NSAIDs_Total_Years = sum(
      EX.NSAIDsDurationYears,
      EX.NSAIDsDurationMonths_Yr,
      EX.NSAIDsDurationWeeks_Yr,
      EX.NSAIDsDurationDays_Yr,
      na.rm = TRUE
    ),
    
    Aspirin_Total_Years = sum(
      EX.AsprinDurationYears,
      EX.AsprinDurationMonths_Yr,
      EX.AsprinDurationWeeks_Yr,
      EX.AsprinDurationDays_Yr,
      na.rm = TRUE
    ), 
    
    PPI_Total_Years = sum(
      EX.ProtonPumpInhibitorDurationYears,
      EX.ProtonPumpInhibitorDurationMonths_Yr,
      EX.ProtonPumpInhibitorDurationWeeks_Yr,
      EX.ProtonPumpInhibitorDurationDays_Yr,
      na.rm = TRUE
    ),
    
    OTCAcidSupp_Total_Years = sum(
      EX.OverTheCounterAcidSuppressantsDurationYears,
      EX.OverTheCounterAcidSuppressantsDurationMonths_Yr,
      EX.OverTheCounterAcidSuppressantsDurationWeeks_Yr,
      EX.OverTheCounterAcidSuppressantsDurationDays_Yr,
      na.rm = TRUE
    ),
    
    H2RA_Total_Years = sum(
      EX.HistamineType2ReceptorAntagonistDurationYears,
      EX.HistamineType2ReceptorAntagonistDurationMonths_Yr,
      EX.HistamineType2ReceptorAntagonistDurationWeeks_Yr,
      EX.HistamineType2ReceptorAntagonistDurationDays_Yr,
      na.rm = TRUE
    ),
    
  ) %>% 
  mutate(NSAIDs = case_when(
  EX.NSAIDs %in% c('never', 'no') ~ 'never',
  EX.NSAIDs %in% c('past use', 'occasional use', 'current use') ~ 'ever'
)) %>%
  mutate(EX.Asprin = case_when(
  EX.Asprin %in% c('never', 'no') ~ 'never',
  EX.Asprin %in% c('past use', 'occasional use', 'current use') ~ 'ever'
)) %>%  
  mutate(EX.ProtonPumpInhibitor = case_when(
  EX.ProtonPumpInhibitor %in% c('never', 'no') ~ 'never',
  EX.ProtonPumpInhibitor %in% c('past use', 'occasional use', 'current use') ~ 'ever'
)) %>%
  mutate(EX.OverTheCounterAcidSuppressants = case_when(
  EX.OverTheCounterAcidSuppressants %in% c('never', 'no') ~ 'never',
  EX.OverTheCounterAcidSuppressants %in% c('past use', 'occasional use', 'current use') ~ 'ever'
)) %>%
  mutate(EX.HistamineType2ReceptorAntagonist = case_when(
  EX.HistamineType2ReceptorAntagonist %in% c('never', 'no') ~ 'never',
  EX.HistamineType2ReceptorAntagonist %in% c('past use', 'occasional use', 'current use') ~ 'ever'
)) %>%
  
  mutate(TAKEN_NSAIDS = case_when(
      NSAIDs %in% c('ever') | EX.Asprin %in% c('ever') ~ "Y",
      TRUE ~ "N"
    ),
    TAKEN_NSAIDS = if_else(is.na(NSAIDs) & is.na(EX.Asprin), "NA", TAKEN_NSAIDS),
    TAKEN_NSAIDS = na_if(TAKEN_NSAIDS, "NA")) %>% 
  
  mutate(NSAIDs_Total_Years_n = na_if(NSAIDs_Total_Years, 0.000000)) %>% 
  mutate(Aspirin_Total_Years_n = na_if(Aspirin_Total_Years, 0.000000)) %>% 
  mutate(PPI_Total_Years_n = na_if(PPI_Total_Years, 0.000000)) %>% 
  mutate(OTCAcidSupp_Total_Years_n = na_if(OTCAcidSupp_Total_Years, 0.000000)) %>% 
  mutate(H2RA_Total_Years_n = na_if(H2RA_Total_Years, 0.000000)) %>% 
  
  mutate(    
    ACID_SUPPRESSANT = case_when(
      EX.ProtonPumpInhibitor %in% c("ever") |
        EX.HistamineType2ReceptorAntagonist %in% c("ever") |
          EX.PatientOnAcidSuppressant %in% c("yes") |
            EX.OverTheCounterAcidSuppressants %in% c("ever") ~ "Y",
      TRUE ~ "N"
    ),
    ACID_SUPPRESSANT = if_else(
      is.na(EX.ProtonPumpInhibitor) &
        is.na(EX.HistamineType2ReceptorAntagonist) &
          is.na(EX.PatientOnAcidSuppressant) &
            is.na(EX.OverTheCounterAcidSuppressants),
      "NA",
      ACID_SUPPRESSANT
    ),
    ACID_SUPPRESSANT = na_if(ACID_SUPPRESSANT, "NA")
  ) %>%
  select(
    -c(
      EX.NSAIDsDurationYears,
      EX.NSAIDsDurationMonths,
      EX.NSAIDsDurationWeeks,
      EX.NSAIDsDurationDays,
      EX.NSAIDsDurationMonths_Yr,
      EX.NSAIDsDurationWeeks_Yr,
      EX.NSAIDsDurationDays_Yr,
      EX.ProtonPumpInhibitorDurationYears,
      EX.ProtonPumpInhibitorDurationMonths,
      EX.ProtonPumpInhibitorDurationWeeks,
      EX.ProtonPumpInhibitorDurationDays,
      EX.ProtonPumpInhibitorDurationMonths_Yr,
      EX.ProtonPumpInhibitorDurationWeeks_Yr,
      EX.ProtonPumpInhibitorDurationDays_Yr,
      EX.OverTheCounterAcidSuppressantsDurationYears,
      EX.OverTheCounterAcidSuppressantsDurationMonths,
      EX.OverTheCounterAcidSuppressantsDurationWeeks,
      EX.OverTheCounterAcidSuppressantsDurationDays,
      EX.OverTheCounterAcidSuppressantsDurationMonths_Yr,
      EX.OverTheCounterAcidSuppressantsDurationWeeks_Yr,
      EX.OverTheCounterAcidSuppressantsDurationDays_Yr,
      EX.HistamineType2ReceptorAntagonistDurationYears,
      EX.HistamineType2ReceptorAntagonistDurationMonths,
      EX.HistamineType2ReceptorAntagonistDurationWeeks,
      EX.HistamineType2ReceptorAntagonistDurationDays,
      PPI_Total_Years,
      OTCAcidSupp_Total_Years,
      H2RA_Total_Years,
      Aspirin_Total_Years,
      NSAIDs_Total_Years
    )
  )
```

## reflux inference

```{r reflux, include=FALSE}
occams_molecular_thesis_cohort_clean_path_epid_meds_reflux <- occams_molecular_thesis_cohort_clean_path_epid_meds %>%
  mutate(heartburn_inferred_bin =
           case_when(
             (
               EX.DurationOfRefluxSymptoms_f %in% c('< 5 years', '5-10 years', '> 10 years', 'unknown') |
                 EX.RefluxFrequencyOfSymptoms_f %in% c("sometimes", "often", "daily", 'unknown') |
                 ACID_SUPPRESSANT == 'Y'
             ) ~ 'Y',
             (
               is.na(EX.DurationOfRefluxSymptoms_f)  &
                 is.na(EX.RefluxFrequencyOfSymptoms_f) &
                 is.na(ACID_SUPPRESSANT)
             ) ~ NA_character_,
             
             TRUE ~ 'N'
           ))
```

# Descriptive stats

```{r desc_data, include=FALSE}

occams_molecular_thesis_cohort_descriptive_statisitics_table_one <- occams_molecular_thesis_cohort_clean_path_epid_meds_reflux %>%
  select(
    ID,
    DNA_ID, 
    abnormal_DNA_ID, 
    normal_DNA_ID,
    cohort,
    
    DI.ageAtDiagnosis,
    age_group, 
    DI.PatientGender,
    Ethnicity, 
    
    BMI_n,
    BMI_Class_f,
    BMI_5yrs_Ago_n,
    BMI_Class_5yrs_Ago_f,
    BMI_Diff_n,
    
    Smoker_Ever,
    smoking_CPD,
    smoking_duration_yrs,
    smoking_NPY, 
    
    EX.HeavyDrinker, 
    EX.UnitsPerWeekInTotal,
    
    EX.DurationOfRefluxSymptoms_f,
    EX.RefluxFrequencyOfSymptoms_f,
    EX.PatientOnAcidSuppressant,
    EX.SymptomaticOnAcidSuppressant,
    EX.ProtonPumpInhibitor,
    PPI_Total_Years_n,
    # PPI_Total_Years,
    EX.OverTheCounterAcidSuppressants,
    OTCAcidSupp_Total_Years_n,
    # OTCAcidSupp_Total_Years,
    EX.HistamineType2ReceptorAntagonist,
    H2RA_Total_Years_n,
    # H2RA_Total_Years,
    ACID_SUPPRESSANT,
    heartburn_inferred_bin,
    
    EX.Asprin,
    Aspirin_Total_Years_n,
    NSAIDs,
    NSAIDs_Total_Years_n,
    TAKEN_NSAIDS,
    
    Phenotype,
    Tumor_Length_CM_n,
    RP.SiewertClassification, 
    RP.TumourDifferentiation.c, 
    RP.Location,
    RP.PostOpPathCombinedTumourSite,
    pTStage_four_f,
    pTNM_f, 
    pTStage_Split_f,
    RP.PS.Tumour.Diff.c,
    RP.Tumour.Growth.c,
    ST.SurgeryPerformed.c,
    
    EX.BarrettsOesophagusUndergoingSurveillance,
    Death, 
    Years_Surv,
    # for alvin ecdna paper nov 2023
    # TR.NeoAdj.ChemotherapyRegimenEpirubicin,
    # TR.NeoAdj.ChemotherapyRegimenCisplatin,
    # TR.NeoAdj.ChemotherapyRegimenOxaliplatin,
    # TR.NeoAdj.ChemotherapyRegimen5FU,
    # TR.NeoAdj.ChemotherapyRegimenCapecitabine,
  ) %>%
  rename(
    # Age = DI.ageAtDiagnosis,
    # Gender = DI.PatientGender,
    # BMI = BMI_n,
    # Obesity = BMI_Obese_f,
    # Smoking = Smoker_Ever,
    # `Reflux Symptoms Frequency` = ,
    # `Reflux Symptoms Duration` = EX.DurationOfRefluxSymptoms_f,
    # `PPI Use` = EX.ProtonPumpInhibitor,
    # `PPI Use Duration (years)` = PPI_Total_Years_n,
    # `OTC Acid Suppres. Use` = EX.OverTheCounterAcidSuppressants,
    # `OTC Acid Suppres. Use Duration (years)` = OTCAcidSupp_Total_Years_n,
    # `NSAID Use` = NSAIDs,
    # `NSAID Use Duration (years)` = NSAIDs_Total_Years_n,
    Phenotype = Phenotype,
    # `Tumor Length (cm)` = Tumor_Length_CM_n,
    # TNM = pTNM_f
  )
```

### save baseline dataset

```{r}
# occams_molecular_thesis_cohort_descriptive_statisitics_table_one
# write_csv(occams_molecular_thesis_cohort_descriptive_statisitics_table_one, 'data/processed/wgs_processed/occams_thesis_wgs_cohort_20230402.csv')
# write_csv(occams_molecular_thesis_cohort_descriptive_statisitics_table_one, 'data/processed/wgs_processed/occams_thesis_wgs_cohort_20240129.csv')
```



```{r desc_table_data, echo=FALSE, fig.align='center'}
occams_molecular_table1_data <-
  # read_csv('data/processed/wgs_processed/occams_thesis_wgs_cohort_20230401.csv')  %>%  
  read_csv('data/processed/wgs_processed/occams_thesis_wgs_cohort_20230402.csv')  %>%  
  # select(-ID) %>%
  # filter(cohort == 'GEL') %>%
  filter(cohort == '710') %>%
  select(Phenotype,
         cohort,
         DI.ageAtDiagnosis,
         age_group,
         DI.PatientGender,
         Ethnicity,
         BMI_Class_f,
         Smoker_Ever,
         TAKEN_NSAIDS, 
         heartburn_inferred_bin,
         EX.UnitsPerWeekInTotal,
         RP.SiewertClassification,
         pTNM_f)
  # %>% filter(EX.BarrettsOesophagusUndergoingSurveillance %in% c('yes'))

occams_molecular_table1 <- table1::table1(~ . | Phenotype,
  data = occams_molecular_table1_data,
  render.continuous = c(. = "Mean [SD]", . = "Median [Q1, Q3]"))
  # render.missing=NULL,
  # render.categorical="FREQ (PCTnoNA%)")
```

```{r}
occams_molecular_table1
```

# Univar regression 

I ran univariable and mutivariable models using the finalfit package - I get the same ORs/CIs using this package or the built in glm() models - just don't use the metrics in the finalfit arguement. I then use the KableExtra package to get column % of variables inclduing missing data. These rest is copy/pasting in Excel

```{r}
dependent <- "Phenotype"
independent <-
  c(
         'DI.ageAtDiagnosis',
         'age_group',
         'DI.PatientGender',
         # 'Ethnicity',
         'BMI_Class_f',
         'Smoker_Ever',
         'TAKEN_NSAIDS', 
         'heartburn_inferred_bin',
         'pTNM_f')

occams_molecular_table1_data %>%
    mutate(BMI_Class_f = case_when(
    BMI_Class_f == "normal" ~ "0_normal",
    BMI_Class_f == "underweight" ~ "0_normal",
    BMI_Class_f == "overweight" ~ "2_overweight",
    BMI_Class_f == "obese" ~ "3_obese"
  )) %>%
  filter(Phenotype %in% c("BE/IM EAC", "Non-BE/IM EAC")) %>%
  mutate(Phenotype = factor(Phenotype, c("BE/IM EAC", "Non-BE/IM EAC"), ordered = TRUE)) %>% 
  mutate(across(.cols = c(2:9), ~ as.character(.x))) %>%
  mutate(across(.cols = c(3:9), ~ replace_na(.x, "z_missing"))) %>%
  finalfit::finalfit(
    dependent,
    independent,
    keep_models = F,
    column = T
  )  %>%
  # select(-c(fit_id, index)) %>%
  kableExtra::kable(row.names = F) %>%
  kableExtra::kable_styling(full_width = F)
```


# For alvin ecdna paper

```{r}

# for alvin re ecdna paper in nov 2023
load('data/original/wgs/710_ecdna/ecDNA-710_surival-analysis.RData') 
any.events <- any.events %>% 
  rename(ID = PatientId) %>% 
  mutate(ID = str_replace(ID, 'OC/', 'OCCAMS/'),)
# use the dataset below
# occams_molecular_thesis_cohort_descriptive_statisitics_table_one

table_one_avlin_ecdna <- inner_join(any.events, occams_molecular_thesis_cohort_descriptive_statisitics_table_one, "ID") %>%
  select(
    1:14,
    Smoker_ever = Smoker_Ever,
    Smoking_pack_years = smoking_NPY,
    Self_reported_heavy_alc_use = EX.HeavyDrinker,
    Symptoms_heartburn_ever = heartburn_inferred_bin,
    Reflux_med_use_ever = ACID_SUPPRESSANT,
    NSAID_med_use_ever = TAKEN_NSAIDS,
    T_stage = pTStage_four_f,
    TR.NeoAdj.ChemotherapyRegimenEpirubicin,
    TR.NeoAdj.ChemotherapyRegimenCisplatin,
    TR.NeoAdj.ChemotherapyRegimenOxaliplatin,
    TR.NeoAdj.ChemotherapyRegimen5FU,
    TR.NeoAdj.ChemotherapyRegimenCapecitabine,
    Death, 
    Years_Surv
  )

write_csv(table_one_avlin_ecdna, 'data/processed/table-1-data_710-cohort.csv')
```

```{r}
library(table1)


table1::table1(
  ~ DI.ageAtDiagnosis+
  age_group+
  DI.PatientGender+
  Ethnicity+
  BMI_n+
  BMI_Class_f+
  Smoker_ever+
  Smoking_pack_years+
  Self_reported_heavy_alc_use+
  Symptoms_heartburn_ever+
  Reflux_med_use_ever+
  NSAID_med_use_ever+
  T_stage | ecDNA,
  data = table_one_avlin_ecdna
)
```

# Figure for paper

## phenotype data
use `occams_molecular_thesis_cohort_descriptive_statisitics_table_one` to generate the phenotype matrix for the oncoplot. 
```{r}
occams_molecular_thesis_cohort_oncoplot <-
  read_csv('data/processed/wgs_processed/occams_thesis_wgs_cohort_20240129.csv') %>%
  select(
    DNA_ID,
    Phenotype,
    cohort,
    age_group,
    DI.PatientGender,
    smoking_NPY,
    Smoker_Ever,
    BMI_n,
    BMI_Class_f,
    TAKEN_NSAIDS,
    heartburn_inferred_bin,
    EX.HeavyDrinker,
    pTStage_Split_f
  ) 
  #####################
  # filter(Phenotype == "BE/IM EAC") %>% 
  # filter(Phenotype == "Unknown BE/IM EAC") %>% 
  # filter(Phenotype == "Non-BE/IM EAC") %>% 
  filter(cohort == "710")
```
## genotype data

use `03_OAC-driver-genes.Rmd` to read the dataframe. 
```{r}
OAC_MPE_SNV_INDEL_CNA_VEP_dataset <-
  read_csv('data/processed/wgs_processed/OAC_MPE_SNV_INDEL_CNA_VEP_dataset_20220217.csv') %>%
  mutate(Phenotype = factor(
    Phenotype,
    levels = c("BE/IM EAC", "Unknown BE/IM EAC", "Non-BE/IM EAC")
  )) %>% arrange(Phenotype) 
# %>% 
  #####################
  # filter(Phenotype == "BE/IM EAC")
  # filter(Phenotype == "Unknown BE/IM EAC")
  # filter(Phenotype == "Non-BE/IM EAC")

# phenotype_info <- OAC_MPE_SNV_INDEL_CNA_VEP_dataset %>% select(DNA_ID, Phenotype)
# phenotype_info$Phenotype_ordered <- factor(phenotype_info$Phenotype, levels = c("BE/IM EAC", "Unknown BE/IM EAC", "Non-BE/IM EAC"))
# ordered_phenotype_info <- phenotype_info[order(phenotype_info$Phenotype_ordered), ]


cna_muts <- OAC_MPE_SNV_INDEL_CNA_VEP_dataset %>% 
  filter(amp_del_value <= -2 | amp_del_value >= 2) %>% 
  filter(parsed_consequence_grouped_c %in% c('CNA_GAIN', 'CNA_LOSS')) %>% 
  select(DNA_ID, gene_name, CNA)


snv_muts <- OAC_MPE_SNV_INDEL_CNA_VEP_dataset %>% 
  filter(parsed_consequence_grouped_c != 'Synonymous') %>% # only take functional muts
  filter(parsed_consequence_grouped_c != 'CNA_NEUTRAL') %>%
  filter(parsed_consequence_grouped_c != 'CNA_GAIN') %>%
  filter(parsed_consequence_grouped_c != 'CNA_LOSS') %>%
  select(DNA_ID, gene_name, alt_type) %>% 
  mutate(alt_type = if_else(alt_type %in% c('ins','del'), 'IND', 'SNV'))


combined_cna_snv_mut_matrix <- bind_rows(cna_muts, snv_muts) %>% 
  mutate(alteration = coalesce(alt_type, CNA)) %>% 
  select(DNA_ID, gene_name, alteration) %>% 
  pivot_wider(names_from = DNA_ID, id_cols = gene_name, values_from = alteration)
 
# ordered_sample_ids <- ordered_phenotype_info$DNA_ID

mut_mat <- as.data.frame(combined_cna_snv_mut_matrix) 
rownames(mut_mat) <- mut_mat[[1]]
mut_mat <- data.frame(lapply(mut_mat, function(x) sapply(x, function(y) paste(y, collapse = ";"))))
mut_mat <- mut_mat[-1] 
mut_mat <- as.matrix(mut_mat)
```



```{r}
library(ComplexHeatmap)
library(tidyverse) 
```

## plot
```{r}
col = c(SNV = "#26A818", IND = "#ffaa00", GAIN = '#EA2E49', LOSS = '#174D9D')

# oncoPrint(mut_mat,
#     alter_fun = list(
#         SNV = function(x, y, w, h) grid.rect(x, y, w*0.9, h*0.9, 
#             gp = gpar(fill = col["SNV"], col = NA)),
#         IND = function(x, y, w, h) grid.rect(x, y, w*0.9, h*0.4, 
#             gp = gpar(fill = col["IND"], col = NA)),
#         LOSS = function(x, y, w, h) grid.rect(x, y, w*0.7, h*0.6, 
#             gp = gpar(fill = col["LOSS"], col = NA)),
#         GAIN = function(x, y, w, h) grid.rect(x, y, w*0.7, h*0.6, 
#             gp = gpar(fill = col["GAIN"], col = NA))
#     ), col = col, )

# Count the number of alterations per gene
alteration_counts <- rowSums(mut_mat != "", na.rm = TRUE)
# Sort and get the names of the top 20 genes
top_genes <- names(sort(alteration_counts, decreasing = TRUE))[1:16]
# Subset the matrix to include only the top 20 genes
top_genes_mat <- mut_mat[top_genes, ]


# BO_pos <-
# BO_unk <-
# BO_neg <-
BO_all <-
oncoPrint(
  top_genes_mat,
  alter_fun = list(
    background = function(x, y, w, h)
      grid.rect(x, y, w * 0.8, h * 0.5, gp = gpar(fill = "#CCCCCC", col = NA)),
    SNV = function(x, y, w, h)
      grid.rect(x, y, w * 1, h * 0.25,
                gp = gpar(fill = col["SNV"], col = NA)),
    IND = function(x, y, w, h)
      grid.rect(x, y, w * 1, h * 0.25,
                gp = gpar(fill = col["IND"], col = NA)),
    LOSS = function(x, y, w, h)
      grid.rect(x, y, w * 1, h * 0.5,
                gp = gpar(fill = col["LOSS"], col = NA)),
    GAIN = function(x, y, w, h)
      grid.rect(x, y, w * 1, h * 0.5,
                gp = gpar(fill = col["GAIN"], col = NA))
  ),
  col = col,
  top_annotation = c(
    HeatmapAnnotation(column_barplot = anno_oncoprint_barplot(height = unit(3, "cm"))),
    phenotype_annotations,
    wgd_annotations,
    tmb_annotations,
    ploidy_annotations
  ),
  bottom_annotation = c(
    age_annotations,
    gender_annotations,
    smoker_annotations,
    bmi_annotatons,
    heartburn_annotations,
    Tstage_annotations
  )
)

# BO_pos
```
## annotations
```{r}

mut_mat_ids <- as_data_frame(names(as_data_frame(mut_mat))) %>% 
  rename(DNA_ID = value) %>% 
  mutate(DNA_ID = str_replace_all(DNA_ID, '\\.', '\\-'))

annotations <- left_join(mut_mat_ids, occams_molecular_thesis_cohort_oncoplot, by = 'DNA_ID') %>% 
  mutate(BMI_Class_f = factor(
    case_when(
      BMI_n < 18.5 ~ "normal",
      BMI_n >= 18.5 & BMI_n <= 24.9 ~ "normal",
      BMI_n >= 25.0 & BMI_n <= 29.9 ~ "overweight",
      BMI_n >= 30.0 ~ "obese"
    ),
    levels = c("normal", "overweight", "obese"),
    ordered = TRUE
  ))

age_group <- 
factor(annotations$age_group, levels = c("< 50 years old", "50 - 59 years old", "60 - 69 years old", "70+ years old"), ordered = T)
age_annotations <-
  HeatmapAnnotation(
    df = data.frame(Age = age_group),
    # col = list(
      # age_group = c(
        # "70+ years old" = "#2C5985",
        # "60 - 69 years old" = "#3480A2",
        # "50 - 59 years old" = '#52A8BC',
        # "< 50 years old" = '#BCE4D8'
      # )
    # ),
    show_legend = TRUE
  )

gender <- factor(annotations$DI.PatientGender, levels = c('male', 'female'), ordered = T)
gender_annotations <-
  HeatmapAnnotation(
    df = data.frame(Gender = gender),
    # col = list(
    #   gender = c(
    #     "male" = "blue",
    #     "female" = "pink"
    #   )
    # ),
    show_legend = TRUE
  )

smoking_ever <- annotations$Smoker_Ever
smoker_annotations <-
  HeatmapAnnotation(
    df = data.frame(`Cigarette smoking` = smoking_ever),
    # col = list(
    #   smoking_ever = c(
    #     "never" = "gray",
    #     "ever" = "#A51122"
    #   )
    # ),
    show_legend = TRUE
  )

bmi_groups <- factor(annotations$BMI_Class_f, levels = c("normal", "overweight", "obese"), ordered = T)
bmi_annotatons <- HeatmapAnnotation(
      df = data.frame(BMI = bmi_groups),
    #   col = list(
    #   gender = c(
    #     "normal" = "#FFBEB2",
    #     "overweight" = "orange",
    #     "obese" = "#A51122"
    #   )
    # ),
    show_legend = TRUE
  )

heartburn_ever <- annotations$heartburn_inferred_bin
heartburn_annotations <-
  HeatmapAnnotation(
    df = data.frame(Heartburn = heartburn_ever),
    # col = list(
    #   df = c(
    #     "N" = "gray",
    #     "Y" = "#A51122"
    #   )
    # ),
    show_legend = TRUE
  )

Tstage_split <- annotations$pTStage_Split_f
Tstage_annotations <-
  HeatmapAnnotation(
    df = data.frame(T.stage = Tstage_split),
    # col = list(
    #   df = c(
    #     "T1/T2" = "#8C7F5F",
    #     "T3/T4" = "#161616"
    #   )
    # ),
    show_legend = TRUE
  )

phenotype_groups <- annotations$Phenotype
phenotype_annotations <- 
  HeatmapAnnotation(
    df = data.frame(Phenotype = phenotype_groups),
    # col = list(
    #   df = c(
    #     "Non-BE/IM EAC" = "red",
    #     "BE/IM EAC" = "blue", 
    #     "Unknown BE/IM EAC" = 'white' 
    #   )
    # ),
    show_legend = TRUE
  )

wgd_yes <- WGD_BO_status$WGD
wgd_annotations <- 
    HeatmapAnnotation(
    df = data.frame(WGD = wgd_yes),
    # col = list(
    #   df = c(
    #     "TRUE" = "black",
    #     "FALSE" = "white"
    #   )
    # ),
    show_legend = TRUE
  )

ploidy_cont <- WGD_BO_status$Ploidy
ploidy_annotations <- 
    HeatmapAnnotation(
    df = data.frame(Ploidy = ploidy_cont),
    show_legend = TRUE
  )
tmb_log10Mb <- OAC_tmb$log10_total_count_per_mb
tmb_annotations <- 
    HeatmapAnnotation(
    df = data.frame(TMB = tmb_log10Mb),
    show_legend = TRUE
  )
```


## genomic events

```{r}
OAC_per_sample_ploidy <- read_tsv('data/original/wgs/710_oac_gistic/per_sample_ploidy.txt') %>% 
  rename(DNA_ID = Sample)



WGD_BO_status <- inner_join(mut_mat_ids, OAC_per_sample_ploidy, "DNA_ID")




OAC_tmb <-
left_join(mut_mat_ids,OAC_tmb <- read_csv('data/processed/wgs_processed/oac_total_snv_indel_TMB_20230401.csv') %>% select(DNA_ID, log10_total_count_per_mb), "DNA_ID")
```



